server/src/main/resources/rules/rules.js

```diff
@@ -502,58 +506,41 @@ function get_pool_priority(pool, consumer) {
         var null_rule_score = 0;
         var mismatch_rule_score = 0;
 
+        log.debug("Evaluating attribute " + attr + " with weight " + attrs[attr]);
         if (unsatisfiedSet.length === 0 && poolSet.length === 0) {
             null_rule_score = 0.1;
+            log.debug("NULL rule score for attribute "+attr+" and pool "+pool.id+", which is: "+null_rule_score);
         }
 
         if (unsatisfiedSet.length > 0) {
             match_rule_score = Utils.intersection(unsatisfiedSet, poolSet).length / unsatisfiedSet.length;
+            log.debug("MATCH rule score for attribute "+attr+" and pool "+pool.id+", which is: "+match_rule_score);
         }
 
         if (specifiedSet.length > 0 && poolSet.length > 0) {
             mismatch_rule_score = (Utils.difference(specifiedSet, poolSet).length / specifiedSet.length) * -0.5;
+            log.debug("MISMATCH rule score for attribute "+attr+" and pool "+pool.id+", which is: "+mismatch_rule_score);
         }
 
         attrScore = (null_rule_score + match_rule_score + mismatch_rule_score) * attrs[attr];
+        log.debug("Final syspurpose score for attribute "+attr+" and pool "+pool.id+" = "+attrScore);
 
-        if (attrScore != 0) {
-            log.debug("evaluating " + attr + " with weight " + attrs[attr]);
-            if (null_rule_score != 0) {
-                log.debug("null rule score: " + null_rule_score);
-            }
-            if (match_rule_score != 0) {
-                log.debug("match rule score: " + match_rule_score);
-                matchesPurpose = true;
-            }
-            if (mismatch_rule_score != 0) {
-                log.debug("mismatch rule score: " + mismatch_rule_score);
-            }
-
-            log.debug("final score = " + attrScore);
-        }
         priority += attrScore;
     });
 
-    // increment to give more weightage to syspurpose fields
-    if (matchesPurpose) {
-        priority += 450;
-        log.debug("incrementing syspurpose score by 450");
-    }
+    log.debug("Current overall score is: "+priority);
 
     // use virt only if possible
     // if the consumer is not virt, the pool will have been filtered out
     if (Utils.equalsIgnoreCase(pool.getProductAttribute(VIRT_ONLY), "true")) {
         priority += 100;
+        log.debug("Increasing score by 100 for attribute: virt_only.");
     }
 
     // better still if host_specific
     if (pool.getAttribute(REQUIRES_HOST_ATTRIBUTE) !== null) {
         priority += 150;
-    }
-
-    // Decrease the priority of shared pools slightly so that non-shared pools will get consumed first.
-    if (pool.hasSharedAncestor) {
-        priority -= 10;
```

Comment:
The removal of this statement has nothing to do with the actual changes for the issue at hand. It is simply done to keep the rules in hosted (candlepin-2.3-HOTFIX) in sync with master/satellite branches.